lib/SearchAndSortPane/SearchAndSortPane.js

```diff
@@ -0,0 +1,350 @@
+import React, {
+  Component,
+  createRef,
+} from 'react';
+import PropTypes from 'prop-types';
+import queryString from 'query-string';
+import { withRouter } from 'react-router';
+import { FormattedMessage } from 'react-intl';
+import {
+  debounce,
+  get,
+  noop,
+} from 'lodash';
+
+import {
+  Pane,
+  Paneset,
+  SRStatus,
+  PaneHeader,
+  Button,
+} from '@folio/stripes/components';
+import {
+  withStripes,
+  stripesShape,
+} from '@folio/stripes/core';
+import {
+  mapNsKeys,
+  getNsKey,
+  makeConnectedSource,
+  buildUrl,
+} from '@folio/stripes/smart-components';
+
+import { SearchForm } from '../SearchForm';
+import { SearchResults } from '../SearchResults';
+import {
+  buildSortOrder,
+  SORT_TYPES,
+} from '../utils';
+
+import css from './SearchAndSortPane.css';
+
+@withRouter
+@withStripes
+export class SearchAndSortPane extends Component {
+  static propTypes = {
+    stripes: stripesShape.isRequired,
+    label: PropTypes.node.isRequired,
+    renderHeader: PropTypes.func,
+    searchLabelKey: PropTypes.string,
+    resultCountMessageId: PropTypes.string.isRequired,
+    location: PropTypes.oneOfType([
+      PropTypes.shape({
+        search: PropTypes.string.isRequired,
+        pathname: PropTypes.string.isRequired,
+      }).isRequired,
+      PropTypes.string.isRequired,
+    ]),
+    history: PropTypes.shape({ // provided by withRouter
+      push: PropTypes.func.isRequired,
+    }).isRequired,
+    match: PropTypes.shape({ // provided by withRouter
+      path: PropTypes.string.isRequired,
+    }).isRequired,
+    parentMutator: PropTypes.shape({
+      query: PropTypes.shape({
+        replace: PropTypes.func.isRequired,
+        update: PropTypes.func.isRequired,
+      }),
+      resultCount: PropTypes.shape({ replace: PropTypes.func.isRequired }).isRequired,
+    }).isRequired,
+    parentResources: PropTypes.shape({
+      query: PropTypes.shape({
```

Comment:
I suggest specifying without details since they are not explicitly used in the component.
```suggestion
      query: PropTypes.ojbect,
```